### PR TITLE
Code documentation

### PR DESCRIPTION
**Pull request by Turtyo**
# Description

Write documentation for the codebase (which can be read from the Godot editor)

## Related issue(s)

Linked with #90 (doesn't close it probably)

## List of changes

- Comment class names
- Comment functions and parameters
- Update the CONTRIBUTING.md to ask for doc in new code

## Additional notes

Draft for now, so people can start reading what i wrote already

### CODE REVIEW

**Comment by Turtyo** [comment]
I'm wondering if I should write a description for every test. I could at least comment the test setups in TestBase or TestMapBase for example. Or I can just comment each test, explaining what we do and what we expect
What do you think @Tysterman74 ?

**Comment by Tysterman74** [comment]
> I'm wondering if I should write a description for every test. I could at least comment the test setups in TestBase or TestMapBase for example. Or I can just comment each test, explaining what we do and what we expect
> What do you think @Tysterman74 ?

I think just the TestBase and TestMapBase should be fine. For the tests themselves the naming of the function should be telling enough as to what the test does since I feel there will be mostly just setup code

**Comment by Turtyo** [comment]
Alright, seems logical to me.
I will comment the setup in details then

**Review by Saphust — DISMISSED** [review]

**Comment by Turtyo** [comment]
Mmh all tests seem to be failing due to NodeJS currently. We'll check that with @Tysterman74 
Checks actually pass on this PR
@Saphust could you approve again please ?

**Review by Saphust — DISMISSED** [review]

**Review by Tomzkk — APPROVED** [review]

File: Entity/Components/HealthComponent.gd [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+## The maximum health the entity can have; this is a hard limit
 @export var max_health: float = 100
-var current_health: float = 100
-
+## The health the entity currently has [br]
+## Between 0 and max_health
+var current_health: float = 100 # ? change this to max health, or just unset since it's init by the ready
 
+## Initialize the health component, putting health to max health
+## This only happens at game start, because this is globally loaded
```
[the remark this answers is not in the document]
Comment (Turtyo) [reply Entity/Components/HealthComponent.gd:17]: Mmh I think I thought I should modify that, let me check what i wrote for the player too
I'm pretty sure that's incorrect

File: Entity/Components/HealthComponent.gd [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
 func _ready() -> void:
 	_set_health(max_health)
 
 
 # Allow caster to be null, but not the target.
 # If caster is null, we assume that the damage came from an unknown source,
 # so status won't calculate.
-# Use a negative damage value if you want to heal.
+## Deal damage to the entity [br]
```
[the remark this answers is not in the document]
Comment (Turtyo) [reply Entity/Components/HealthComponent.gd:25]: I'm not sure it works, I think doc comments must be just before the body of the function
Let me check that

[the remark this answers is not in the document]
Comment (Turtyo) [reply Entity/Components/HealthComponent.gd:25]: As expected, it doesn't work, I moved the one # comments inside the function instead

**Review by Tomzkk — APPROVED** [review]

**Review by Saphust — APPROVED** [review]

**Pull request merged**